stackstate_checks_base/stackstate_checks/base/utils/http_helper.py

```diff
@@ -0,0 +1,1447 @@
+                                To fix this you can either modify the schematic validation or remove it entirely
+                                """.format(str(self._validate_schematic), e))
+
+            except ValueError as e:
+                errors.append("""Unable to parse the response as JSON.
+                                 The response received was {0}.
+                                 Full error from the JSON parse attempt {1}
+                                 """.format(str(response.content), e))
+
+        # If the HTTP Response Validation Status Code has been set
+        if self._validate_status_code is not None and response.status_code != self._validate_status_code:
+            errors.append("""The response was unable to conform to the validation status code. The status code
+                             applied was {0}
+                             The expected status code is {1}
+                             To fix this you can either modify the status code validation or remove it entirely
+                             """.format(str(response.status_code), str(self._validate_status_code)))
+        # If the status code has not been set then we will by default see 400 > as bad responses.
+        elif response.status_code >= 400:
+            errors.append("""Server responded with a {0}""".format(str(response.status_code)))
+
+        # If the HTTP Response Type has been set
+        # Test if the HTTP Response Type is a enum of type HTTPResponseType
+        if self._validate_type is not None and \
+                type(self._validate_type) is HTTPResponseType and \
+                isinstance(self._validate_type, Enum) is True:
+
+            # If the type set is JSON. We then attempt to parse it and see if it is valid
+            if self._validate_type is HTTPResponseType.JSON:
+                try:
+                    data = json.loads(response.content.decode(response.encoding))
+                    if isinstance(data, dict) is False:
+                        raise ValueError()
+
+                except ValueError as e:
+                    errors.append("""Unable to parse the response as JSON.
+                                     The response received was {0}.
+                                     Full error from the JSON parse attempt {1}
+                                     """.format(str(response.content), e))
+
+            elif isinstance(self._validate_type.value, type(response.content)) is False:
+                errors.append("""The response content type does not conform to the validation type. The response
+                                 type is {0}.
+                                 The expected type is {1}
+                                 To fix this you can either modify the type validation or remove it entirely
+                                 """.format(str(type(response.content)), str(self._validate_type.value)))
+
+        # Return all the errors that occurred
+        return errors
```

Comment:
In all other places you raise exceptions, but all of a sudden here you return errors? I would say that by this point you might as well just raise an exception here with an exception with a message that contains all errors.

Reply:
With the previous version you mentioned to not throw on the response and rather allow the developer to handle it. With the above I am compile a list of things that went wrong instead of throwing a error